components/scream/scripts/gather_all_data.py

```diff
@@ -50,20 +51,19 @@ def formulate_command(self, machine):
 
         repo = scream_docs_repo if self._scream_docs else scream_repo
 
-        # Compute kokkos location
-        if self._kokkos:
-            kokkos_loc = pathlib.Path(self._kokkos.replace("$machine", machine).replace("$compiler", compiler))
-        else:
-            kokkos_loc = pathlib.Path(scream_repo.parent.parent, "externals", "kokkos")
-
         local_cmd = self._run
         # Do magic replacements here
         local_cmd = local_cmd.\
-            replace("$compiler", compiler).\
-            replace("$kokkos", str(kokkos_loc)).\
+            replace("-cxx $cxx_compiler", cxx_compiler).\
+            replace("-f90 $f90_compiler", f90_compiler).\
             replace("$machine", machine).\
             replace("$scream_docs", str(scream_docs_repo)).\
             replace("$scream", str(scream_repo))
+        if self._kokkos:
```

Comment:
It seems like it might be easier to do the kokkos stuff before the other replacements so you don't have to repeat them.